### PR TITLE
Dont try to update an instance that isnt running yet

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

There is a bug where we can try to update an instance that hasn't technically started running yet - this fixes it to not update unless we are running

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_app/utilities/app_helpers.py [hunk 1]
```diff
@@ -556,7 +556,10 @@ def _handle_is_headless(app: "LightningApp"):
             "App was not found. Please open an issue at https://github.com/lightning-AI/lightning/issues."
         )
 
-    if current_lightningapp_instance.spec.is_headless == app.is_headless:
+    if any(
+        current_lightningapp_instance.spec.is_headless == app.is_headless,
+        current_lightningapp_instance.status.phase != V1LightningappInstanceState.RUNNING,
+    ):
```
Comment (ethanwharris) [inline src/lightning_app/utilities/app_helpers.py:562]:
```suggestion
    if any([
        current_lightningapp_instance.spec.is_headless == app.is_headless,
        current_lightningapp_instance.status.phase != V1LightningappInstanceState.RUNNING,
    ]):
```

File: src/lightning_app/core/app.py [hunk 2]
```diff
@@ -526,14 +526,11 @@ def _update_layout(self) -> None:
...
-        is_headless = _is_headless(self)
+        self.is_headless = _is_headless(self)
 
         # If `is_headless` changed, handle it.
         # This ensures support for apps which dynamically add a UI at runtime.
-        if self.is_headless != is_headless:
-            self.is_headless = is_headless
-
-            _handle_is_headless(self)
+        _handle_is_headless(self)
```
Comment (hhsecond) [inline src/lightning_app/core/app.py:533]: why did we remove the guard here, without that wouldn't we making non necessary requests to the be?

Comment (ethanwharris) [reply src/lightning_app/core/app.py:533]: No, I had two guards by accident - we already have something in place that says we won't make a request unless the value changes

Comment (hhsecond) [reply src/lightning_app/core/app.py:533]: cool

**Review by hhsecond — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**